network/dell/dnos10_config.py

```diff
@@ -0,0 +1,260 @@
+        and when set to false, the remote device will not be updated.
+    required: false
+    default: yes
+    choices: ['yes', 'no']
+  backup_config:
+    description:
+      - This argument will cause the module to create a full backup of
+        the current C(running-config) from the remote device before any
+        changes are made.  The backup file is written to the C(backup)
+        folder in the playbook root directory.  If the directory does not
+        exist, it is created.
+    required: false
+    default: no
+    choices: ['yes', 'no']
+"""
+
+EXAMPLES = """
+- dnos10_config:
+    lines: ['hostname {{ inventory_hostname }}']
+    force: yes
+
+- dnos10_config:
+    lines:
+      - 10 permit ip host 1.1.1.1 any log
+      - 20 permit ip host 2.2.2.2 any log
+      - 30 permit ip host 3.3.3.3 any log
+      - 40 permit ip host 4.4.4.4 any log
+      - 50 permit ip host 5.5.5.5 any log
+    parents: ['ip access-list extended test']
+    before: ['no ip access-list extended test']
+    match: exact
+
+- dnos10_config:
+    lines:
+      - 10 permit ip host 1.1.1.1 any log
+      - 20 permit ip host 2.2.2.2 any log
+      - 30 permit ip host 3.3.3.3 any log
+      - 40 permit ip host 4.4.4.4 any log
+    parents: ['ip access-list extended test']
+    before: ['no ip access-list extended test']
+    replace: block
+
+- dnos10_config:
+    commands: "{{lookup('file', 'datcenter1.txt')}}"
+    parents: ['ip access-list test']
+    before: ['no ip access-list test']
+    replace: block
+
+"""
+
+RETURN = """
+updates:
+  description: The set of commands that will be pushed to the remote device
+  returned: always
+  type: list
+  sample: ['...', '...']
+
+responses:
+  description: The set of responses from issuing the commands on the device
+  retured: when not check_mode
+  type: list
+  sample: ['...', '...']
+"""
+from ansible.module_utils.netcfg import NetworkConfig, dumps, ConfigLine
+from ansible.module_utils.dnos10 import NetworkModule, dnos10_argument_spec
+from ansible.module_utils.dnos10 import get_config, get_sublevel_config 
+
+def get_candidate(module):
+    candidate = NetworkConfig(indent=1)
+    if module.params['src']:
+        candidate.load(module.params['src'])
+    elif module.params['lines']:
+        parents = module.params['parents'] or list()
+        candidate.add(module.params['lines'], parents=parents)
+    return candidate
+
+
+def main():
+
+    argument_spec = dict(
+        lines=dict(aliases=['commands'], type='list'),
+        parents=dict(type='list'),
+
+        src=dict(type='path'),
+
+        before=dict(type='list'),
+        after=dict(type='list'),
+
+        match=dict(default='line',
+                   choices=['line', 'strict', 'exact', 'none']),
+        replace=dict(default='line', choices=['line', 'block']),
+
+        update_config=dict(type='bool', default=True),
+        backup_config=dict(type='bool', default=False)
```

Comment:
to remain consistent with other network *_config modules please rename argument to ```backup```.  You can add an alias to ```backup_config``` if you want to use it